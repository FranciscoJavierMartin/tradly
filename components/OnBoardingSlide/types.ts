export interface Slide {
  text: string;
  img: string;
}
