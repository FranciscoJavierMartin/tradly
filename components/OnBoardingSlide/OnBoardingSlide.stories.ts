import OnBoardingSlide from './OnBoardingSlide.vue';
import slideMocks from '../OnBoarding/mock';

export default {
  title: 'OnBoardingSlide',
  component: OnBoardingSlide,
  args: {
    slide: slideMocks[1],
    index: 0,
    amountOfSlides: 3,
  },
};

const Template = (args) => ({
  components: { OnBoardingSlide },
  setup() {
    return { args };
  },
  template: '<OnBoardingSlide v-bind="args"/>',
});

export const Basic = Template.bind({});
Basic.args = {};
