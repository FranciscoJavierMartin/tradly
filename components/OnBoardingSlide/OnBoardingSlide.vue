<template>
  <div class="c-on-boarding-slide">
    <img :src="slide.img" :alt="slide.text" />
    <h3>{{ slide.text }}</h3>
    <div class="c-on-boarding-slide__steps">
      <div
        v-for="index in amountOfSlides"
        :key="index"
        :class="{
          'c-on-boarding-slide__steps--active': currentIndex === index,
        }"
      />
    </div>
    <button @click="onClick">
      {{ currentIndex === amountOfSlides ? 'Finish' : 'Next' }}
    </button>
  </div>
</template>

<script setup lang="ts">
import { PropType } from 'vue';
import { Slide } from './types';

const props = defineProps({
  slide: {
    type: Object as PropType<Slide>,
    required: true,
  },
  currentIndex: {
    type: Number,
    required: true,
  },
  amountOfSlides: {
    type: Number,
    required: true,
  },
});

const emit = defineEmits(['next']);

function onClick(): void {
  emit('next');
}
</script>

<style lang="scss">
@import '../../assets/styles/main.scss';

.c-on-boarding-slide {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: 15px;
  row-gap: 15px;

  img {
    background-color: $white;
    border-radius: 8px;
    padding: 40px 5px 5px;
    width: 70%;
  }

  h3 {
    color: $dark-green;
    font-size: 20px;
    font-weight: 500;
    letter-spacing: -0.482353px;
    line-height: 30px;
    text-align: center;
  }

  &__steps {
    display: flex;

    > div {
      background: $light-green;
      border-radius: 50%;
      height: 12px;
      margin: 5px;
      mix-blend-mode: normal;
      opacity: 0.6;
      width: 12px;

      &.c-on-boarding-slide__steps--active {
        background-color: $dark-green;
        opacity: 1;
      }
    }
  }

  button {
    background-color: $dark-green;
    border-radius: 24px;
    border: none;
    color: $white;
    font-family: $monserrat-font;
    font-size: 18px;
    font-weight: 600;
    line-height: 18px;
    padding: 15px;
    text-align: center;
    text-decoration: none;
    width: 100%;
  }
}
</style>
