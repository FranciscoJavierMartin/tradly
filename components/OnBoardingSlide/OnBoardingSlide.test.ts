import '@testing-library/jest-dom';
import { describe, expect, test } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/vue';
import OnBoardingSlide from './OnBoardingSlide.vue';
import slideMocks from '../OnBoarding/mock';

describe('OnBoardingSlide.vue', () => {
  test('should render', () => {
    const { container } = render(OnBoardingSlide, {
      props: {
        slide: slideMocks[0],
        currentIndex: 1,
        amountOfSlides: 3,
      },
    });

    expect(
      screen.getByRole('heading', { name: slideMocks[0].text })
    ).toBeInTheDocument();
    expect(screen.getByAltText(slideMocks[0].text)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should be the first step be highlight', () => {
    render(OnBoardingSlide, {
      props: {
        slide: slideMocks[0],
        currentIndex: 1,
        amountOfSlides: 3,
      },
    });

    const activeElements = document.querySelectorAll(
      '.c-on-boarding-slide__steps--active'
    );
    expect(activeElements.length).toBe(1);
  });

  test('should show "Next" on the button when item is not the last', () => {
    render(OnBoardingSlide, {
      props: {
        slide: slideMocks[0],
        currentIndex: 1,
        amountOfSlides: 3,
      },
    });

    expect(screen.getByRole('button', { name: /Next/i })).toBeInTheDocument();
  });

  test('should show "Next" on the button when item is not the last', () => {
    render(OnBoardingSlide, {
      props: {
        slide: slideMocks[0],
        currentIndex: 3,
        amountOfSlides: 3,
      },
    });

    expect(screen.getByRole('button', { name: /Finish/i })).toBeInTheDocument();
  });

  test('Click on the button should trigger "next" event', async () => {
    const { emitted } = render(OnBoardingSlide, {
      props: {
        slide: slideMocks[0],
        currentIndex: 1,
        amountOfSlides: 3,
      },
    });

    await fireEvent.click(screen.getByText('Next'));
    expect(emitted()).toHaveProperty('next');
  });
});
