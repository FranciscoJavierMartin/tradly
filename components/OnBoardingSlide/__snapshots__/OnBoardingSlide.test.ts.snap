// Vitest Snapshot v1

exports[`OnBoardingSlide.vue > should render 1`] = `
<div
  class="c-on-boarding-slide"
>
  <img
    alt="Empowering Artisans, Farmers & Micro Business"
    src="../../assets/imgs/on-boarding-step-1.svg"
  />
  <h3>
    Empowering Artisans, Farmers & Micro Business
  </h3>
  <div
    class="c-on-boarding-slide__steps"
  >
    
    <div
      class="c-on-boarding-slide__steps--active"
    />
    <div
      class=""
    />
    <div
      class=""
    />
    
  </div>
  <button
    class="c-custom-button"
  >
    Next
  </button>
</div>
`;
